contrib_bots/bots_test_lib.py

```diff
@@ -41,13 +44,40 @@ def mock_test(self, messages, message_handler, bot_response):
         with patch('contrib_bots.bot_lib.BotHandlerApi') as MockClass:
             instance = MockClass.return_value
 
-            for (message, response) in zip(messages, bot_response):
-                # Send message to the concerned bot
-                message_handler.handle_message(message, MockClass(), StateHandler())
+            with patch('requests.get') as mock_get:
+                mock_result = mock.MagicMock()
+                mock_result.json.return_value = http_response
+                mock_result.ok.return_value = True
+                mock_get.return_value = mock_result
+
+                for (message, response) in zip(messages, bot_response):
+                    # Send message to the concerned bot
+                    message_handler.handle_message(message, MockClass(), StateHandler())
+                    # Check if the bot is sending a reply message.
+                    # 'response' is a string here.
+                    instance.send_reply.assert_called_with(message, response)
+
+    def mock_test_send_message(self, messages, message_handler, bot_response, http_request=None, http_response=None):
+        # message_handler is of type Any, since it can contain any bot's
+        # handler class. Eventually, we want bot's handler classes to
+        # inherit from a common prototype specifying the handle_message
+        # function.
+        # type: (List[Dict[str, Any]], Any, Dict[str, str]) -> None
+        # Mocking BotHandlerApi
+        with patch('contrib_bots.bot_lib.BotHandlerApi') as MockClass:
+            instance = MockClass.return_value
 
-                # Check if BotHandlerApi is sending a reply message.
-                # This can later be modified to assert the contents of BotHandlerApi.send_message
-                instance.send_reply.assert_called_with(message, response)
+            with patch('requests.get') as mock_get:
+                mock_result = mock.MagicMock()
+                mock_result.json.return_value = http_response
+                mock_result.ok.return_value = True
+                mock_get.return_value = mock_result
+                for (message, response) in zip(messages, bot_response):
+                    # Send message to the concerned bot
+                    message_handler.handle_message(message, MockClass(), StateHandler())
+                    # Check if the bot is sending a message via `send_message` function.
+                    # Where response is a dictionary here.
+                    instance.send_message.assert_called_with(response)
```

Comment:
We should be able to deduplicate most of the code between this and `mock_test_send_reply` by moving this block into a function.